specs/proposals.md

```diff
@@ -24,15 +30,39 @@ are part of later specification milestones.
 
 [cannon]: https://github.com/ethereum-optimism/cannon
 
-## Proposing L2 output commitments
+## Definitions
 
-The proposer's role is to construct and submit output commitments on a configurable interval to a contract on L1.
+### Constants
 
-TODO: describe integration with rollup node and L2 execution engine (see PR #179).
+| Name                   | Value | Unit    |
+| ---------------------- | ----- | ------- |
+| `SUBMISSION_INTERVAL` | `1800` | seconds |
+| `L2_BLOCK_TIME`        | `2`   | seconds |
 
-TODO: link to contract specification/source of L2 Output oracle on L1.
+### Types
 
-## L2 output commitment construction
+The `ForkSpec` type contains the height and blockhash for a block in the L1 chain.
+
+```js
+struct ForkSpec {
+    uint256 blockHeight;
+    bytes32 blockHash;
+}
+```
+
+## Proposing L2 Output Commitments
+
+The proposer's role is to construct and submit output commitments on a configurable interval to a contract on , which
+it does by running the [L2 output submitter](../l2os/) (AKA L2OS). This service periodically queries the rollup
```

Comment:
When I hear L2OS i think of "layer 2 operating system"

Reply:
Same. WDYT about L2OSS, for output submitter service?
Otherwise open to suggestions.

Reply:
Personally I like to avoid acronyms unless it's really cumbersome or already in common use, so we could just use "L2 output submitter" everywhere. It is annoying for the package name, but there my suggestion is `l2output`.

Not a hill I'm going to die on, but I suggest discussing this on a call someday, reaching a decision and moving on.

Reply:
L2OSS -> L2 Open Source Software lol

L2 output submitter submitter I think is nice, also not opposed to just using what we use now (batch submitter)

Reply:
Going to handle this for now by just removing the "AKA" note.